### DISCUSSION TITLE
add deprecation warning for using/initialising structs too

### DISCUSSION THREAD

**Discussion by i582** [vlang/v#18692] [Ideas]
### Describe the bug

Code: https://vosca.dev/p/fcc60e55bf

```v
[deprecated: 'some']
struct Some {}

println(Some{})

```

### Expected Behavior

Warning about using of deprecated `Some`

### Current Behavior

Output:

```
Some{}
```

### Reproduction Steps

Run code above

### V version

V 0.3.4 fc4c431.45f16a2

### Environment details (OS name and version, etc.)

```
V full version: V 0.3.4 fc4c431.34ac326
OS: linux, Ubuntu 22.04.2 LTS
Processor: 2 cpus, 64bit, little endian, Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz

getwd: /home/pmakhnev/playground
vexe: /home/pmakhnev/v/v
vexe mtime: 2023-06-25 00:00:09

vroot: OK, value: /home/pmakhnev/v
VMODULES: OK, value: /root/.vmodules
VTMP: OK, value: /tmp/v_0

Git version: git version 2.34.1
Git vroot status: weekly.2023.25-30-g34ac3269 (2 commit(s) behind V master)
.git/config present: true

CC version: cc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```

**Comment by spytheman** [comment]
The `[deprecated]` tags are currently implemented only for function declarations, and for struct field declarations.

**Comment by i582** [comment]
> The `[deprecated]` tags are currently implemented only for function declarations, and for struct field declarations.

ah, I see, thank you!
